instrumentation/okhttp3/testing/src/androidTest/java/io/opentelemetry/instrumentation/library/okhttp/v3_0/InstrumentationTest.kt

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+@file:Suppress("ktlint:standard:package-name")
+
+package io.opentelemetry.instrumentation.library.okhttp.v3_0
+
+import io.opentelemetry.android.test.common.OpenTelemetryRumRule
+import io.opentelemetry.api.OpenTelemetry
+import io.opentelemetry.api.trace.Span
+import io.opentelemetry.exporter.otlp.http.trace.OtlpHttpSpanExporter
+import io.opentelemetry.sdk.OpenTelemetrySdk
+import io.opentelemetry.sdk.trace.SdkTracerProvider
+import io.opentelemetry.sdk.trace.export.SimpleSpanProcessor
+import mockwebserver3.MockResponse
+import mockwebserver3.MockWebServer
+import okhttp3.Call
+import okhttp3.Callback
+import okhttp3.Interceptor
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.Response
+import org.junit.After
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import java.io.IOException
+import java.util.concurrent.CountDownLatch
+
+class InstrumentationTest {
+    private lateinit var server: MockWebServer
+
+    @get:Rule
+    internal var openTelemetryRumRule: OpenTelemetryRumRule = OpenTelemetryRumRule()
+
+    @Before
+    @Throws(IOException::class)
+    fun setUp() {
+        server = MockWebServer()
+        server.start()
+    }
+
+    @After
+    @Throws(IOException::class)
+    fun tearDown() {
+        server.close()
+    }
+
+    @Test
+    @Throws(IOException::class)
+    fun okhttpTraces() {
+        server.enqueue(MockResponse.Builder().code(200).build())
+
+        val span = openTelemetryRumRule.getSpan()
+
+        span.makeCurrent().use { ignored ->
+            val client =
+                OkHttpClient
+                    .Builder()
+                    .addInterceptor(
+                        Interceptor { chain: Interceptor.Chain? ->
+                            val currentSpan = Span.current().spanContext
+                            Assert.assertEquals(
+                                span.spanContext.traceId,
+                                currentSpan.traceId,
```

Comment:
It was already structured this way, so you're not making it worse or anything, but any time I see assertions inside of callback/async/conditional code I'm leery, because there can be unexpected cases where the callback (in this case an interceptor) doesn't get invoked at all and the test still passes without the assertion being invoked.

If we're going to keep that around, then I think there also needs to be a side effect that we can also assert on to ensure that the assertion was invoked (I've frequently used countdown latches for such a thing).